### ISSUE TITLE
Add a link to Pootle or Pontoon on strings

### ISSUE THREAD

**Issue by flodolo** [mozfr/transvision#823]
It would be great to have a link to open the editor (Pootle, Pontoon) directly, like we link to sources.

**Comment by SkySymbol** [comment]
Hello, I would like to work on this :-)

For example, for this search

https://transvision.mozfr.org/?sourcelocale=en-US&locale=es-AR&repo=aurora&search_type=entities&recherche=browser/chrome/browser/aboutDialog.dtd:update.failed.linkText&perfect_match=perfect_match

we need a link like 
https://pontoon.mozilla.org/es-AR/firefox-aurora/browser/chrome/browser/aboutDialog.dtd/?search=Descargar+la+última+versión

Is that correct ? Do we have strings that are not in Pontoon ?

**Comment by flodolo** [comment]
It would be safer to use string IDs.

For example, for Pontoon: https://pontoon.mozilla.org/es-AR/firefox-aurora/all-resources/?search=update.failed.linkText&string=76057

For Pootle: https://mozilla.locamotion.org/gn_PY/firefox/translate/#search=update.failed.linkText&sfields=locations

A few challenges:
* We should only show this for Aurora, because we don't have other repositories. This would soon change.
* Figure out where to show that information without cluttering the UI.
* Keep a list of locales working on Pootle, and locales working on Pontoon.

**Comment by jotes** [comment]
Hi,
I think that Pontoon-side blocked by following issues:
* https://bugzilla.mozilla.org/show_bug.cgi?id=1305281
* https://bugzilla.mozilla.org/show_bug.cgi?id=1302053

As @mathjazz mentioned at irc channel, you can use a /?search= querystring but that won't be the perfect solution. If you have any suggestions/requests  - I'll try to help.

**Comment by julen** [comment]
> For Pootle: https://mozilla.locamotion.org/gn_PY/firefox/translate/#search=update.failed.linkText&sfields=locations

If the path is at hand, it's safe to assume this pattern will work:
```
https://<server.tld>/<locale_code>/<project_code>/translate/<file-path>.po#search=<string-id>&sfields=locations
```

This way the results are _a bit_ more immediate, since the scope is reduced to a single file as opposed to the entire project.

**Comment by SkySymbol** [comment]
Yes i'm using links like this one for pontoon : https://pontoon.mozilla.org/es-AR/firefox-aurora/all-resources/?search=update.failed.linkText

Thank you :)

**Comment by flodolo** [comment]
I believe searching for the string ID would enough for now, even if it shows multiple results (not a great chance given that you search for the identifier, not the text).

**Comment by SkySymbol** [comment]
Perfect @julen

**Comment by mathjazz** [comment]
> https://bugzilla.mozilla.org/show_bug.cgi?id=1305281
> https://bugzilla.mozilla.org/show_bug.cgi?id=1302053

Actually I don't think that would help much.

However, I'd suggest using the URL with the file path (instead of `all-resources`) if it's not too hard, in which case you'll almost never hit multiple results.

**Comment by SkySymbol** [comment]
I think we should be able to generate links like this : https://pontoon.mozilla.org/es-AR/firefox-aurora/browser/chrome/browser/aboutDialog.dtd/?search=Descargar+la+última+versión

I'll ask here or on irc if I have questions. Thanks for the help!

**Comment by SkySymbol** [comment]
Two questions:

1- Is it a safe assumption to say that if a locale is not present in Pootle, it's present in Pontoon, or the reverse ?

- Not every locale seems to have a firefox-aurora project on Pontoon / Pootle, does the code have to keep a track of where the 'firefox-aurora' project is for each locale?

**Comment by flodolo** [comment]
Not at the moment, I'll need to find a way to provide that.

In the meantime, for fixing this PR, I suggest to hardcode a couple of lists with a few locales, we can improve it later.

**Comment by StoyanDimitrov** [comment]
Links to `mail` and `mobile` folders to Pontoon are not good. They open in Firefox Aurora project.

**Comment by flodolo** [comment]
> Links to mail and mobile folders to Pontoon are not good. They open in Firefox Aurora project.

Good point, filing a new issue
